### ISSUE TITLE
Tunnel simulation sometimes too late (GPS signal lost)

### ISSUE THREAD

**Issue by scaidermern** [osmandapp/Osmand#5879]
Sometimes the tunnel simulation starts too late, leading to the GPS signal getting lost (which then only continues after the tunnel). This happens especially in hilly terrain where the GPS accuracy is low.

Is it possible to add/increase a buffer before tunnels? Alternatively if the GPS signal is lost, check if we are right before a tunnel.

**Comment by gunnar22** [comment]
I agree. I sometimes drive on a nearby road with seven tunnels. There is a lot of "GPS signal lost" and "GPS signal recovered" messages from the navigation. I have been thinking that these messages are just noise because everyone knows that there is no GPS signals inside the tunnel. And OsmAnd also knows that it is navigating into a tunnel, so shouldn't it be possible to suppress such messages? Would be nice.

**Comment by XandrexOSM** [comment]
in the current voice navigation options, there is a setting for "how much in advance" the navigation hint will occur (see screenshot below).

Maybe this can be enhanced in the specific case when the next change is inside a tunnel, then make it occur before entering the tunnel?
with an option on/off for that.

![Screenshot_2023_09_03_13_49_55_32_014aeb6b5e04ee7837dcdaecc5244511](https://github.com/osmandapp/OsmAnd/assets/92858966/62c08f00-6d21-4539-b243-686d2a2e98e1)
